File: java/src/main/java/com/adaptionsoft/games/trivia/RandomInput.java
package com.adaptionsoft.games.trivia;

import java.util.Random;

public class RandomInput implements Input {
    private final Random random = new Random();

    @Override
    public int die() {
        return random.nextInt(6) + 1;
    }

    @Override
    public boolean responseIsCorrect() {
        return random.nextInt(10) != 7;
    }
}

File: kotlin/src/main/kotlin/com/adaptionsoft/games/trivia/RandomInput.kt
package com.adaptionsoft.games.trivia

import java.util.*

class RandomInput : Input {
    private val rand = Random()
    
    override fun die(): Int {
        return rand.nextInt(5) + 1
    }
    
    override fun responseIsCorrect(): Boolean {
        return rand.nextInt(9) != 7
    }
}